test/ZapV2.test.ts

```diff
@@ -0,0 +1,223 @@
+import { ethers, network, waffle } from "hardhat"
+import { expect } from "chai"
+import { BigNumber } from "ethers"
+import { WAVAX } from "@traderjoe-xyz/sdk"
+
+const JOE_ADDRESS = "0x6e84a6216eA6dACC71eE8E6b0a5B7322EEbC0fDd"
+const WAVAX_ADDRESS = "0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7"
+const USDT_ADDRESS = "0xc7198437980c041c805A1EDcbA50c1Ce5db95118"
+const USDC_ADDRESS = "0xA7D7079b0FEaD91F3e65f86E8915Cb59c1a4C664"
+const APEX_ADDRESS = "0xd039c9079ca7f2a87d632a9c0d7cea0137bacfb5"
+
+const ROUTER_ADDRESS = "0x60aE616a2155Ee3d9A68541Ba4544862310933d4"
+
+const JOEAVAX_ADDRESS = "0x454E67025631C065d3cFAD6d71E6892f74487a15"
+const AVAXAPEX_ADDRESS = "0x824Ca83923990b91836ea927c14C1fb1B1790B08"
+const USDCUSDT_ADDRESS = "0x2E02539203256c83c7a9F6fA6f8608A32A2b1Ca2"
+
+describe("zapV2", function () {
+  before(async function () {
+    // ABIs
+    this.zapV2CF = await ethers.getContractFactory("ZapV2")
+    this.joeMakerCF = await ethers.getContractFactory("JoeMaker")
+    this.ERC20CF = await ethers.getContractFactory("JoeERC20")
+    this.PairCF = await ethers.getContractFactory("JoePair")
+    this.RouterCF = await ethers.getContractFactory("JoeRouter02")
+
+    // Account
+    this.signers = await ethers.getSigners()
+    this.dev = this.signers[0]
+    this.alice = this.signers[1]
+
+    // Contracts
+    this.router = await this.RouterCF.attach(ROUTER_ADDRESS, this.alice)
+
+    // Tokens
+    this.wavax = await ethers.getContractAt("IWAVAX", WAVAX_ADDRESS)
+    this.wavax_erc20 = await this.ERC20CF.attach(WAVAX_ADDRESS)
+    this.joe = await this.ERC20CF.attach(JOE_ADDRESS)
+    this.usdt = await this.ERC20CF.attach(USDT_ADDRESS)
+    this.usdc = await this.ERC20CF.attach(USDC_ADDRESS)
+    this.apex = await this.ERC20CF.attach(APEX_ADDRESS)
+
+    // Pairs
+    this.joeAvax = await this.PairCF.attach(JOEAVAX_ADDRESS)
+    this.usdcUsdt = await this.PairCF.attach(USDCUSDT_ADDRESS)
+    this.avaxApex = await this.PairCF.attach(AVAXAPEX_ADDRESS)
+  })
+
+  beforeEach(async function () {
+    // We reset the state before each tests
+    await network.provider.request({
+      method: "hardhat_reset",
+      params: [
+        {
+          forking: {
+            jsonRpcUrl: "https://api.avax.network/ext/bc/C/rpc",
+            blockNumber: 8104212,
+          },
+          live: false,
+          saveDeployments: true,
+          tags: ["test", "local"],
+        },
+      ],
+    })
+
+    // We redeploy zapV2 for each tests too
+    this.zapV2 = await this.zapV2CF.deploy(ROUTER_ADDRESS)
+    await this.zapV2.deployed()
+  })
+
+  describe("Revert", function () {
+    it("Revert on zapIn", async function () {
+      //AVAX was at ~$83.05, so this will revert (as there is the 0.3 swap fees)
+      await expect(
+        this.zapV2
+          .connect(this.alice)
+          .zapIn("0", "41500000", [WAVAX_ADDRESS, USDT_ADDRESS, USDC_ADDRESS], [WAVAX_ADDRESS, USDC_ADDRESS, USDT_ADDRESS], {
+            value: "1000000000000000000",
```

Comment:
nit: you can use `ethers.utils.parseEther("1")` so its a bit easier to understand how much this is w/o counting the zeros. same suggestion for the rest of this file as well